mainnet/gateways/lua.mjs

```diff
@@ -0,0 +1,134 @@
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import fs from 'fs';
+import path from 'path';
+
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const argv = yargs(hideBin(process.argv))
+  .option('dryRun', {
+    alias: 'd',
+    type: 'boolean',
+    description: 'Run without making any changes',
+    default: false,
+  })
+  .option('output', {
+    alias: 'o',
+    type: 'string',
+    description: 'Output CSV file path',
+    default: './outputs/gateways.lua',
+  })
+  .option('input', {
+    alias: 'i',
+    type: 'string',
+    description: 'Input CSV file path',
+    default: './outputs/gateways.csv',
+  })
+  .option('startTimestamp', {
+    alias: 's',
+    type: 'number',
+    description: 'Start timestamp (defaults to 2025-02-20 00:00:00 UTC)',
+    default: 1740009600000,
+  })
+  .help()
+  .parseSync();
+
+const dryRun = argv.dryRun;
+const output = path.join(__dirname, argv.output);
+const input = path.join(__dirname, argv.input);
+const startTimestamp = argv.startTimestamp; // the gateways starting timestamp
+console.log(
+  'Creating raw lua gateways with assigned delegates from',
+  input,
+  'and writing to',
+  output,
+);
+
+// mkdir if not exists
+fs.mkdirSync(path.dirname(output), { recursive: true });
+
+// overwrite the file if it exists
+fs.writeFileSync(output, '');
+
+// read all the records from the input file
+// gatewayAddress,observerAddress,operatorStake,fqdn,port,protocol,allowDelegatedStaking,delegateRewardShareRatio,minDelegatedStake,autoStake,label,note,properties,status,failedConsecutiveEpochsCount
+const gateways = fs
+  .readFileSync(input, 'utf8')
+  .split('\n')
+  .slice(1)
+  .map((row) => row.trim())
+  .filter((row) => row.length > 0)
+  .filter(Boolean) // Remove empty lines
+  .map((row) => row.split(','));
+
+for (const gateway of gateways) {
+  const [
+    gatewayAddress,
+    observerAddress,
+    operatorStake,
+    fqdn,
+    port,
+    protocol,
+    allowDelegatedStaking,
+    delegateRewardShareRatio,
+    minDelegatedStake,
+    autoStake,
+    label,
+    note,
+    properties = 'FH1aVetOoulPGqgYukj0VE0wIhDy90WiQoV3U2PeY44',
+    status,
+    failedConsecutiveEpochsCount,
+  ] = gateway;
+  const luaRecord = `
+GatewayRegistry["${gatewayAddress}"] = {
+    observerAddress = "${observerAddress}",
+    operatorStake = ${operatorStake},
+    totalDelegatedStake = 0,
+    startTimestamp = ${startTimestamp},
+    settings = {
+        fqdn = "${fqdn}",
+        port = ${port},
+        note = "${note}",
+        label = "${label}",
+        protocol = "${protocol}",
+        properties = "${properties}",
+        minDelegatedStake = ${minDelegatedStake},
+        allowDelegatedStaking = ${allowDelegatedStaking},
+        delegateRewardShareRatio = ${delegateRewardShareRatio},
+        autoStake = ${autoStake},
+        allowedDelegates = {},
```

Comment:
so existing delegates and the allowed delegates setting are not being migrated?

Reply:
not as of now, this may be modified. there is no UI that allowed anyone to set them, so anyone that did it would've had to programatically

Reply:
delegates makes sense, as they may no longer have mainnet balance